src/main/flight/imu.c

```diff
@@ -178,49 +155,36 @@ void imuResetAccelerationSum(void)
 }
 
 #if defined(USE_ALT_HOLD)
-static void imuTransformVectorBodyToEarth(t_fp_vector * v)
-{
-    // From body frame to earth frame
-    const float x = rMat[0][0] * v->V.X + rMat[0][1] * v->V.Y + rMat[0][2] * v->V.Z;
-    const float y = rMat[1][0] * v->V.X + rMat[1][1] * v->V.Y + rMat[1][2] * v->V.Z;
-    const float z = rMat[2][0] * v->V.X + rMat[2][1] * v->V.Y + rMat[2][2] * v->V.Z;
-
-    v->V.X = x;
-    v->V.Y = -y;
-    v->V.Z = z;
-}
-
 // rotate acc into Earth frame and calculate acceleration in it
 static void imuCalculateAcceleration(timeDelta_t deltaT)
 {
-    static int32_t accZoffset = 0;
+    static float accZoffset = 0;
     static float accz_smooth = 0;
 
     // deltaT is measured in us ticks
     const float dT = (float)deltaT * 1e-6f;
 
-    t_fp_vector accel_ned;
-    accel_ned.V.X = acc.accADC[X];
-    accel_ned.V.Y = acc.accADC[Y];
-    accel_ned.V.Z = acc.accADC[Z];
-
-    imuTransformVectorBodyToEarth(&accel_ned);
+    quaternion accel_ned;
+    accel_ned.x = acc.accADC[X];
+    accel_ned.y = acc.accADC[Y];
+    accel_ned.z = acc.accADC[Z];
+    quaternionTransformVectorBodyToEarth(&accel_ned, &qAttitude);
 
     if (imuRuntimeConfig.acc_unarmedcal == 1) {
         if (!ARMING_FLAG(ARMED)) {
             accZoffset -= accZoffset / 64;
```

Comment:
this is actually first order low-pass filter. Float implementation should use RC constant ...